src/components/Sections/SectionsProgress.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react'
+import { Button, Grid, Icon, List, Progress } from 'semantic-ui-react'
+import { Progress as ProgressType, SectionAndPage, SectionsStructureNEW } from '../../utils/types'
+import strings from '../../utils/constants'
+
+interface SectionsProgressProps {
+  sections: SectionsStructureNEW
+  resumeApplication: (location: SectionAndPage) => void
+}
+
+const SectionsProgress: React.FC<SectionsProgressProps> = ({ sections, resumeApplication }) => {
+  const getIndicator = ({ completed, valid }: ProgressType) => {
+    return completed ? (
+      <Icon name={valid ? 'check circle' : 'exclamation circle'} color={valid ? 'green' : 'red'} />
+    ) : (
+      <Icon name="circle outline" />
+    )
+  }
+
+  const getProgress = ({ doneRequired, doneNonRequired, totalSum, valid }: ProgressType) => {
+    const totalDone = doneRequired + doneNonRequired
+    return totalDone > 0 && totalSum > 0 ? (
+      <Progress
+        style={{ width: 200 }}
+        percent={(100 * totalDone) / totalSum}
+        size="tiny"
+        success={valid}
+        error={!valid}
+      />
+    ) : null
+  }
+
+  // TODO: Use correct firstIncomplete sections
+  const firstIncomplete: SectionAndPage = { sectionCode: 'S1', pageName: 'Page1' }
+
+  return (
+    <List
+      divided
+      relaxed="very"
+      items={Object.entries(sections).map(([sectionCode, { details, progress }]) => ({
+        key: `list-item-${sectionCode}`,
+        icon: progress ? getIndicator(progress) : null,
+        header: (
+          <Grid stackable style={{ minHeight: 50 }}>
+            <Grid.Column style={{ minWidth: 100 }} floated="left" width={4}>
+              <p>{details.title}</p>
+            </Grid.Column>
+            <Grid.Column style={{ width: 200, paddingRight: '5%' }} floated="right" width={4}>
+              {progress && getProgress(progress)}
+            </Grid.Column>
+            <Grid.Column style={{ minWidth: 100, padding: 0 }} width={2}>
+              {firstIncomplete && sectionCode === firstIncomplete.sectionCode && (
+                <Button color="blue" onClick={() => resumeApplication(firstIncomplete)}>
+                  {strings.BUTTON_APPLICATION_RESUME}
+                </Button>
```

Comment:
Just wondering about the placement of the button. The way it is now, it looks like it's tied to Section 1, which is a bit misleading. (esp since Section 1 is completed in this case):
![Screen Shot 2021-02-26 at 10 38 27 AM](https://user-images.githubusercontent.com/5456533/109223058-f9c94000-781e-11eb-8833-3c0092bf84d3.png)

Maybe place it on its own line below the last section?

Reply:
It's just showing the button in here because the `firstIncomplete` not filled out correctly yet. So it would display in Section 1 for any application. Until I get the changes on master and update this to use the correct one...